File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
enum Encodings {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white,
}

type Color = keyof typeof Encodings

export function decodedValue(colors: Color[]): number {
  return Number(
    colors
      .slice(0, 2)
      .map(color => Encodings[color])
      .join('')
  )
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
from enum import IntEnum, auto


class Color(IntEnum):
    BLACK = 0
    BROWN = auto()
    RED = auto()
    ORANGE = auto()
    YELLOW = auto()
    GREEN = auto()
    BLUE = auto()
    VIOLET = auto()
    GREY = auto()
    WHITE = auto()


def value(colors: list[str]) -> int:
    c1, c2, *_ = colors
    return int(f"{Color[c1.upper()].value}{Color[c2.upper()].value}")
